hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/TestGetContentSummary.java

```diff
@@ -0,0 +1,204 @@
+    checkContentSummary(contentSummary, directories.length, fileCount,
+        dirsWithNonEmptyFiles.length * filesPerDirectory * testBufferSize);
+  }
+
+  @Test
+  public void testFileContentSummary() throws IOException {
+    ContentSummary contentSummary = fs.getContentSummary(pathToFile);
+    checkContentSummary(contentSummary, 0, 1, testBufferSize);
+  }
+
+  @Test
+  public void testLeafDir() throws IOException {
+    ContentSummary contentSummary = fs.getContentSummary(pathToLeafDir);
+    checkContentSummary(contentSummary, 0, 0, 0);
+  }
+
+  @Test
+  public void testIntermediateDirWithFilesOnly() throws IOException {
+    ContentSummary contentSummary =
+        fs.getContentSummary(pathToIntermediateDirWithFilesOnly);
+    checkContentSummary(contentSummary, 0, filesPerDirectory,
+        testBufferSize * filesPerDirectory);
+  }
+
+  @Test
+  public void testIntermediateDirWithFilesAndSubdirs() throws IOException {
+    ContentSummary contentSummary =
+        fs.getContentSummary(pathToIntermediateDirWithFilesAndSubdirs);
+    checkContentSummary(contentSummary, 3, 3 * filesPerDirectory,
+        testBufferSize * filesPerDirectory);
+  }
+
+  @Test
+  public void testDirOverListMaxResultsItems()
+      throws IOException {
+    checkContentSummary(
+        fs.getContentSummary(pathToListMaxDir), 1,
+        numFilesForListMaxTest + filesPerDirectory, 0);
+  }
+
+  @Test
+  public void testInvalidPath() throws Exception {
+    intercept(IOException.class, () -> fs.getContentSummary(new Path(
+        "/nonExistentPath")));
+  }
+
+  @Test
+  public void testConcurrentGetContentSummaryCalls()
+          throws InterruptedException, ExecutionException {
+    ExecutorService executorService = new ThreadPoolExecutor(1,
+            maxThreads, 5, TimeUnit.SECONDS, new SynchronousQueue<>());
+    ArrayList<Future<ContentSummary>> futures = new ArrayList<>();
+    for (String directory : directories) {
+      Future<ContentSummary> future = executorService.submit(
+              () -> fs.getContentSummary(new Path(directory)));
+      futures.add(future);
+    }
+    int[][] dirCS = {{8, 8 * filesPerDirectory, 8 * testBufferSize}, {0, filesPerDirectory, 2 * testBufferSize},
+            {2, 3 * filesPerDirectory, 2 * testBufferSize}, {3, 3 * filesPerDirectory, 2 * testBufferSize},
+            {2, numFilesForListMaxTest + 2 * filesPerDirectory, 0}, {0, filesPerDirectory, 0},
+            {0, filesPerDirectory, filesPerDirectory * testBufferSize}, {1, filesPerDirectory, 0},
+            {0, filesPerDirectory, 0}, {0, 0, 0}, {1, numFilesForListMaxTest + 2, 0}, {0, filesPerDirectory, 0}};
+    executorService.shutdown();
+    for (int i=0; i<directories.length; i++) {
+      ContentSummary contentSummary = futures.get(i).get();
+      checkContentSummary(contentSummary, dirCS[i][0], dirCS[i][1], dirCS[i][2]);
+    }
+  }
+
+  private void checkContentSummary(ContentSummary contentSummary,
```

Comment:
long directoryCount, long fileCount, long byteCount
rename to
long expectedDirectoryCount, long expectedFileCount, long expectedByteCount

Reply:
done